oshmem/include/shmem.h.in

```diff
@@ -207,6 +218,581 @@ extern shmem_ctx_t oshmem_ctx_default;
+OSHMEM_DECLSPEC  int shmem_ulong_alltoall(shmem_team_t team, unsigned long *target, const unsigned long *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_ulonglong_alltoall(shmem_team_t team, unsigned long long *target, const unsigned long long *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_longdouble_alltoall(shmem_team_t team, long double *target, const long double *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_int8_alltoall(shmem_team_t team, int8_t *target, const int8_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_int16_alltoall(shmem_team_t team, int16_t *target, const int16_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_int32_alltoall(shmem_team_t team, int32_t *target, const int32_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_int64_alltoall(shmem_team_t team, int64_t *target, const int64_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_uint8_alltoall(shmem_team_t team, uint8_t *target, const uint8_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_uint16_alltoall(shmem_team_t team, uint16_t *target, const uint16_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_uint32_alltoall(shmem_team_t team, uint32_t *target, const uint32_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_uint64_alltoall(shmem_team_t team, uint64_t *target, const uint64_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_size_alltoall(shmem_team_t team, size_t *target, const size_t *source, size_t nelems);
+OSHMEM_DECLSPEC  int shmem_ptrdiff_alltoall(shmem_team_t team, ptrdiff_t *target, const ptrdiff_t *source, size_t nelems);
+#if OSHMEM_HAVE_C11
+#define shmem_alltoall(...)                                            \
+    _Generic(&*(__OSHMEM_VAR_ARG1(__VA_ARGS__)),                       \
+            shmem_team_t: _Generic(&*(__OSHMEM_VAR_ARG2(__VA_ARGS__)), \
+                char*:        shmem_char_alltoall,                     \
+                short*:       shmem_short_alltoall,                    \
+                int*:         shmem_int_alltoall,                      \
+                long*:        shmem_long_alltoall,                     \
+                long long*:   shmem_longlong_alltoall,                 \
+                signed char*:        shmem_schar_alltoall,             \
+                unsigned char*:      shmem_uchar_alltoall,             \
+                unsigned short*:     shmem_ushort_alltoall,            \
+                unsigned int*:       shmem_uint_alltoall,              \
+                unsigned long*:      shmem_ulong_alltoall,             \
+                unsigned long long*: shmem_ulonglong_alltoall,         \
+                float*:       shmem_float_alltoall,                    \
+                double*:      shmem_double_alltoall,                   \
+                long double*: shmem_longdouble_alltoall,               \
+                default:      __oshmem_datatype_ignore))(__VA_ARGS__)
```

Comment:
may be use same macro as for ```pshmem_test_some_vector ``` to simplify code:
```
#define shmem_ptrdiff_alltoall(team, target, source, nelems)\
     _Generic(&*(target),                                                     \
                 char*:        shmem_char_alltoall,                     \
                 short*:       shmem_short_alltoall,                    \
                 int*:         shmem_int_alltoall,                      \
                 long*:        shmem_long_alltoall,                     \
                 long long*:   shmem_longlong_alltoall,                 \
                 signed char*:        shmem_schar_alltoall,             \
                 unsigned char*:      shmem_uchar_alltoall,             \
                 unsigned short*:     shmem_ushort_alltoall,            \
                 unsigned int*:       shmem_uint_alltoall,              \
                 unsigned long*:      shmem_ulong_alltoall,             \
                 unsigned long long*: shmem_ulonglong_alltoall,         \
                 float*:       shmem_float_alltoall,                    \
                 double*:      shmem_double_alltoall,                   \
                 long double*: shmem_longdouble_alltoall,               \
                 default:      __oshmem_datatype_ignore)(team, target, source, nelems)
```
and same for other macro below

Reply:
Sure will do